server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -404,28 +406,10 @@ public RepositoryData withGenId(long newGeneration) {
     }
 
     /**
-     * Make a copy of this instance with the given UUID and all other fields unchanged.
-     */
-    public RepositoryData withUuid(String uuid) {
```

Comment:
No point exposing this IMO. This can just be set as an implementation detgail if a cluster uuid is set for the first time, no need to expose it separately.